Allow users to add comments on each others profile pages

lenary opened the pull request.

razielgn commented:
Working on it!

razielgn committed:
Experimenting with comments on each user page

razielgn committed:
Added avatar to comments

lenary committed:
smaller avatar

razielgn committed:
Completed the work on comments.

Now a user is able to post comments on another user's page and delete comments on his page.

razielgn commented:
It's ready to be merged.
Specs: https://gist.github.com/1198360
Is it ok?

razielgn committed:
Fixed insecure actions on CommentsController.

+tests

razielgn commented:
About comments, soon I'll integrate a presenter to keep things clean and consistent.

razielgn committed:
Merged master c97c61700999bc2478c447370129ae918b47c64c

razielgn committed:
Added CommentPresenter

razielgn committed:
Added CommentsController#edit and validations.

Now users are able to edit comments on their page or comments the left
on other users' pages. A validation has been introduced so that no empty
or too short comments can be submitted.

razielgn committed:
Refactored CommentsController to use before_filters

razielgn committed:
Fixed comments' css positioning

elight commented:
Looking at the user/show template, I get the sense that there is more that can be extracted.  That is, there's quite a bit of new logic after line 36 in the show.html.haml.  Try extracting more of that into the presenter.

elight commented:
Oh hrm. The Comment's author is still a BSON::ObjectId?  That's probably safer for if/when we switch away form Mongoid.  But I don't believe that's the Mongoid way.

elight commented:
Also, I should only be able to delete comments from my own profile.  At the moment, I can edit them.  Right now, if you were to leave me a comment, I could make it say that "Evan is a just a giant shitbag" when you said such nice things about me. ;-)

razielgn committed:
Improved readability of user/show and fixed rights

Now a user cannot edit comments posted on his own page.

razielgn commented:
About the comment's author being stored as an ObjectId: I really tried to achieve the result _the mongoid way_ but there is just no way to do a 1 to N relation from a document to an embedded one (without storing the id of the comment in the author's doc).

razielgn committed:
Removed a piece of logic from users/show.

elight commented:
Oh, right, the embedding adds complexity. Understood. Further reminder as to why we should consider switching off of Mongoid to just use mongo directly then.

Mongoid seems to limit access to mongodb features while making it easier to use a subset of them. I'd rather have direct mongodb access. 

Evan

On Sep 22, 2011, at 10:10 AM, Federico Ravasioreply@reply.github.com wrote:

> About the comment's author being stored as an ObjectId: I really tried to achieve the result _the mongoid way_ but there is just no way to do a 1 to N relation from a document to an embedded one (without storing the id of the comment in the author's doc).
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/rubypair/rubypair/pull/19#issuecomment-2167499
